core/src/main/java/hudson/cli/declarative/CLIRegisterer.java

```diff
@@ -195,10 +201,33 @@ public int main(List<String> args, Locale locale, InputStream stdin, PrintStream
                                     sc.setAuthentication(old); // restore
                                 }
                             } catch (CmdLineException e) {
-                                stderr.println(e.getMessage());
-                                printUsage(stderr,parser);
-                                return 1;
+                                stderr.println("\nERROR: " + e.getMessage());
+                                printUsage(stderr, parser);
+                                return 2;
+                            } catch (IllegalStateException e) {
+                                stderr.println("\nERROR: " + e.getMessage());
+                                return 4;
+                            } catch (IllegalArgumentException e) {
+                                stderr.println("\nERROR: " + e.getMessage());
+                                return 3;
+                            } catch (AbortException e) {
+                                stderr.println("\nERROR: " + e.getMessage());
+                                return 5;
+                            } catch (AccessDeniedException e) {
+                                stderr.println("\nERROR: " + e.getMessage());
+                                return 6;
+                            } catch (BadCredentialsException e) {
+                                // to the caller, we can't reveal whether the user didn't exist or the password didn't match.
+                                // do that to the server log instead
+                                String id = UUID.randomUUID().toString();
+                                LOGGER.log(Level.INFO, "CLI login attempt failed: " + id, e);
+                                stderr.println("\nERROR: Bad Credentials. Search the server log for " + id + " for more details.");
+                                return 7;
                             } catch (Exception e) {
+                                final String errorMsg = String.format("Unexpected exception occurred while performing %s command.",
+                                        getName());
+                                stderr.println("\nERROR: " + errorMsg);
+                                LOGGER.log(Level.WARNING, errorMsg, e);
                                 e.printStackTrace(stderr);
                                 return 1;
```

Comment:
see [comments in the JIRA ticket](https://issues.jenkins-ci.org/browse/JENKINS-32273?focusedCommentId=246524&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-246524) for a discusion on why we think 1 is not the best choice here.